### PR TITLE
Fix template meta file generation for asset retargetting

### PR DESCRIPTION
**Pull request by chrisfromwork**
This review contains a few changes:

1) Generating meta files during asset retargetting now utilizes the new FileTemplate and TemplatedWriter logic.
2) Player meta file template files have been updated to use *_TEMPLATE_START and *_TEMPLATE_END suffixes.
3) I have removed dated Utilities functions that had previously been used for generating these meta files.

### CODE REVIEW

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/AssetScriptReferenceRetargeter.cs [hunk 1]
```diff
@@ -444,44 +447,37 @@ private static void CopyFiles(DirectoryInfo directory, string pluginPath, string
...
-                    List<string> entries = new List<string>();
-                    foreach (KeyValuePair<string, int> pair in executionOrderEntries)
-                    {
-                        entries.Add(Utilities.ReplaceTokens(executionOrderEntryTemplateBody, new Dictionary<string, string>()
-                        {
-                            { "<SCRIPT_FULL_NAME_TOKEN>", pair.Key },
-                            { "<SCRIPT_EXECUTION_VALUE_TOKEN>", pair.Value.ToString() }
-                        }));
-                    }
+                TemplatedWriter executionOrderWriter = writer.CreateWriterFor("EXECUTION_ORDER");
```
Comment (andreiborodin) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/AssetScriptReferenceRetargeter.cs]: You could single- line these:
```csharp
writer.CreateWriterFor("EXECUTION_ORDER")
   .Write("EMPTY", "{}");
```

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/AssetScriptReferenceRetargeter.cs [hunk 2]
```diff
@@ -444,44 +447,37 @@ private static void CopyFiles(DirectoryInfo directory, string pluginPath, string
...
+            else
+            {
+                TemplatedWriter executionOrderWriter = writer.CreateWriterFor("EXECUTION_ORDER");
+                executionOrderWriter.Write("EMPTY", string.Empty);
 
-                    tokenReplacements.Add(executionOrderTemplate, executionOrderTemplateBody.Replace("<EMPTY_TOKEN>", string.Empty));
-                    tokenReplacements.Add(executionOrderEntryTemplate, string.Join("\r\n", entries));
+                foreach (KeyValuePair<string, int> pair in executionOrderEntries)
+                {
+                    TemplatedWriter executionOrderEntryWriter = writer.CreateWriterFor("EXECUTION_ORDER_ENTRY");
```
Comment (andreiborodin) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/AssetScriptReferenceRetargeter.cs]: These can also be single-chained:
```csharp
writer.CreateWriterFor("EXECUTION_ORDER_ENTRY")
   .Write("SCRIPT_FULL_NAME", pair.Key)
   .Write("SCRIPT_EXECUTION_VALUE", pair.Value.ToString());
```

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/MSBuildTemplates/Plugin.iOS.meta.template [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
   externalObjects: {}
   serializedVersion: 2
   iconMap: {}
-#EXECUTION_ORDER_TEMPLATE   executionOrder: <EMPTY_TOKEN>
-#EXECUTION_ORDER_ENTRY_TEMPLATE     <SCRIPT_FULL_NAME_TOKEN>: <SCRIPT_EXECUTION_VALUE_TOKEN>
+#EXECUTION_ORDER_TEMPLATE_START
+  executionOrder: <EMPTY_TOKEN>
+#EXECUTION_ORDER_TEMPLATE_END
+#EXECUTION_ORDER_ENTRY_TEMPLATE_START
+    <SCRIPT_FULL_NAME_TOKEN>: <SCRIPT_EXECUTION_VALUE_TOKEN>
```
Comment (keveleigh) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/MSBuildTemplates/Plugin.iOS.meta.template]: Not sure if this is intentional or not, but this line has four spaces while the ones above (`iconMap`, `executionOrder`, etc) all have two.

Comment (andreiborodin) [reply Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/MSBuildTemplates/Plugin.iOS.meta.template]: That is intentional, it's a list from above :D

**Pull request merged**